### PR TITLE
Refactor ExactIntegral and ExactNumeric

### PR DESCRIPTION
**Pull request by aslesarenko**
This renaming is to clarify semantics of ExactNumeric.rem method. 
The new name differentiate it from existing methods `mod`, `rem` etc.
Also more ScalaDocs added.

### CODE REVIEW

File: common/src/main/scala/scalan/ExactIntegral.scala [hunk 1]
```diff
@@ -2,54 +2,59 @@ package scalan
...
-  override def toFloat(x: T): Float = n.toFloat(x)
-  override def toDouble(x: T): Double = n.toDouble(x)
-  override def compare(x: T, y: T): Int = n.compare(x, y)
+trait ExactIntegral[T] extends ExactNumeric[T] {
+  protected val n: Integral[T]
+
+  /** Integer division operation `x / y`. */
+  def quot(x: T, y: T): T = n.quot(x, y)
+
+  /** Operation which returns reminder from dividing x by y.
```
Comment (kushti) [inline common/src/main/scala/scalan/ExactIntegral.scala:23]: reminder => remainder

File: core/src/main/scala/scalan/primitives/NumericOps.scala [hunk 2]
```diff
@@ -95,7 +95,7 @@ case class IntegralMod[T](i: ExactIntegral[T])(implicit elem: Elem[T]) extends D
       * numeric types.
       * @see sigmastate.eval.NumericOps.BigIntIsIntegral
       */
-    override def applySeq(x: T, y: T): T = i.rem(x, y)
+    override def applySeq(x: T, y: T): T = i.divisionRemainder(x, y)
```
Comment (kushti) [inline core/src/main/scala/scalan/primitives/NumericOps.scala:98]: ScalaDoc looks outdated, also, fix "reminder" => "remainder" there

**Review by kushti — APPROVED** [review]
LGTM but please check comments

File: common/src/main/scala/scalan/ExactIntegral.scala (hunk 1 above)
Comment (aslesarenko) [reply common/src/main/scala/scalan/ExactIntegral.scala]: fixed here and in all other places

File: core/src/main/scala/scalan/primitives/NumericOps.scala (hunk 2 above)
Comment (aslesarenko) [reply core/src/main/scala/scalan/primitives/NumericOps.scala:75]: fixed

**Pull request merged**